#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_use_libxml2_instead_of_libxml.dpatch by Daniel Leidert
## <daniel.leidert@wgdd.de>
##
## DP: Make use of libxml2 instead of libxml

@DPATCH@
diff -urNad xml2-0.4~/2xml.c xml2-0.4/2xml.c
--- xml2-0.4~/2xml.c	2003-12-24 21:28:48.000000000 +0100
+++ xml2-0.4/2xml.c	2008-03-12 09:32:17.345380803 +0100
@@ -24,7 +24,7 @@
 #include <assert.h>
 #include <ctype.h>
 
-#include <HTMLparser.h>
+#include <libxml/HTMLparser.h>
 
 int do_html;
 int in_tag = 0;
@@ -104,7 +104,7 @@
 			fputs("/>",stdout);
 		else {
 			const htmlElemDesc *elem = NULL;
-			if (do_html) elem = htmlTagLookup(name);
+			if (do_html) elem = htmlTagLookup((xmlChar *) name);
 			finish_tag();
 			if (NULL == elem || (!elem->endTag && !elem->empty)) {
 				fputs("</",stdout);
diff -urNad xml2-0.4~/Makefile.am xml2-0.4/Makefile.am
--- xml2-0.4~/Makefile.am	2008-02-07 16:58:13.000000000 +0100
+++ xml2-0.4/Makefile.am	2008-03-12 09:32:17.345380803 +0100
@@ -1,9 +1,9 @@
-AM_CPPFLAGS = $(XML_CFLAGS)
+AM_CFLAGS = $(XML_CFLAGS)
 bin_PROGRAMS = xml2 2xml csv2 2csv
 xml2_SOURCES = xml2.c
-xml2_LDADD = -lxml
+xml2_LDADD = $(XML_LIBS)
 2xml_SOURCES = 2xml.c
-2xml_LDADD = -lxml
+2xml_LDADD = $(XML_LIBS)
 csv2_SOURCES = csv2.c
 2csv_SOURCES = 2csv.c
 
diff -urNad xml2-0.4~/configure.ac xml2-0.4/configure.ac
--- xml2-0.4~/configure.ac	2008-02-07 17:11:54.000000000 +0100
+++ xml2-0.4/configure.ac	2008-03-12 09:32:17.345380803 +0100
@@ -11,7 +11,7 @@
 AC_PROG_CC
 
 # Checks for libraries.
-PKG_CHECK_MODULES(XML, libxml)
+PKG_CHECK_MODULES(XML, libxml-2.0)
 AC_SUBST(XML_LIBS)
 AC_SUBST(XML_CFLAGS)
 
diff -urNad xml2-0.4~/xml2.c xml2-0.4/xml2.c
--- xml2-0.4~/xml2.c	2001-10-28 05:29:46.000000000 +0100
+++ xml2-0.4/xml2.c	2008-03-12 09:32:17.345380803 +0100
@@ -24,8 +24,8 @@
 #include <string.h>
 #include <ctype.h>
 
-#include <parser.h>
-#include <HTMLparser.h>
+#include <libxml/parser.h>
+#include <libxml/HTMLparser.h>
 
 struct node
 {
